### PR TITLE
Resolve uncovered CUDA dnn layer

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

* Gelu activation layer on CUDA
* Try to relax GEMM from ONNX

resolves https://github.com/opencv/opencv/issues/24064

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/layers/mvn_layer.cpp [hunk 1]
```diff
@@ -350,32 +353,34 @@ class MVNLayerImpl CV_FINAL : public MVNLayer
...
                 double alpha = 1;
                 if (normVariance)
                 {
                     alpha = 1 / std::sqrt(eps + dev[0]*dev[0]);
                 }
-                double normalizationScale = 1.0;
-                double normalizationShift = 0.0;
-                if (fuse_batch_norm)
+
+                if (scale.empty() || scale.total() == newRows)
```
Comment (dkurt) [inline modules/dnn/src/layers/mvn_layer.cpp:364]: Need to verify that such check is not weak. In example, for input `1x2x2`, and weights `1x2`.

**Comment by dkurt** [comment]
Seems like `LayerNorm` changes became bigger and wider. Make sense to move to a separate PR.

**Comment by dkurt** [comment]
@WanliZhong, please review a part about GEMM. I have switched `is_matmul` to `false` when weights are just 2D.

**Comment by asmorkalov** [comment]
@dkurt Thanks a lot for the patch. I made experiments with the code. The matmul change is covered by accuracy tests and it works well, but not covered with performance tests. Could you add some performance test to ensure performance regressions before merge.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by dkurt** [comment]
@asmorkalov, added performance test. To compare, replace `lp.set("is_matmul", weights.dims > 2);` to `lp.set("is_matmul", true);` on the same branch. This is the only change made in ONNX importer.

**Review by fengyuentau — COMMENTED** [review]
Gemm part looks good to me. IIRC, `is_matmul` is added by Wanli and used to deal with some optimization things. @WanliZhong Could you add some details here?

**Comment by dkurt** [comment]
@fengyuentau, `is_matmul` is not about optimization but enabling matmul broadcast. It utilizes the same implementation: https://github.com/opencv/opencv/pull/22828
https://github.com/opencv/opencv/blob/0245c0cd10387f2de7c7bd076585a285d977e53c/modules/dnn/src/layers/fully_connected_layer.cpp#L534-L567

**Review by WanliZhong — APPROVED** [review]
When the matrix is 2D, the implementation is the same whether or not `is_matmul` is true. Someday matrix multiplication should have to be separated from the inner product implementation. :-)

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**